container/layer_tools.bzl

```diff
@@ -185,7 +185,7 @@ def _add_join_layers_go_args(args, inputs, images):
         inputs += image["blobsum"]
 
         if image.get("legacy"):
-            args.add(image["legacy"].path, "--tarball=%s")
+            args.add("--tarball", image["legacy"])
```

Comment:
iirc, I had some tests complain that image["legacy"] was a DirectDependency type and not a File type, but if tests are passing here then this is better.

Reply:
Yes tests pass. I don't think this particular line was previously tested as it failed without my change.